### PR TITLE
Redux implementation

### PR DESCRIPTION
**Pull request by sarthak-sps**
Added redux for state management. (For screens- add transaction, transactions, dashboard)
Made UI components dynamic.

### CODE REVIEW

File: src/component/AccountExpense.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * 
+ * @param {Object} props
+ * @param {number} props.totalExpense - The total expense amount.
+ */
+const AccountExpensesStatus = ({ totalExpense }) => {
+    return (
+        <View style={styles.accountExpensesContainer}>
+            <View style={styles.accountStatusRow}>
+                <View style={styles.statusIconContainer}>
+                    <Image style={styles.statusIcon} source={require('../../assets/images/expense.png')} />
```
Comment (msssoftprodigy008) [inline src/component/AccountExpense.js:17]: Please move all image imports to a single file.

images.js
eg: const expense = require("path to image")

in component - 
import {expense} from 'path_image_file'

File: src/component/UpperComponent.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        <View style={styles.upperContainer}>
+            <View style={styles.headerContainer}>
+                <View>
+                    <Text style={styles.headerLeftText}>{currentDay} {currentDateNumber} </Text>
+                    <Text>{currentMonth}</Text>
+                </View>
+
+                <View style={styles.profileContainer}>
+                    <View style={styles.imageWrapper}>
+                        <Image style={styles.profileImage} source={require('../../assets/images/applogo.png')} />
```
Comment (vaibhavsps1) [inline src/component/UpperComponent.js:40]: import this from utils/images file

File: src/component/AccountIncomeStatus.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ * 
+ * @param {Object} props
+ * @param {number} props.totalIncome - The total income amount.
+ */
+const AccountIncomeStatus = ({ totalIncome }) => {
+    return (
+        <View style={styles.accountIncomeContainer}>
+            <View style={styles.accountStatusRow}>
+                <View style={styles.statusIconContainer}>
+                    <Image style={styles.statusIcon} source={require('../../assets/images/incomeArrow.png')} />
```
Comment (msssoftprodigy008) [inline src/component/AccountIncomeStatus.js:17]: Please move all image imports to a single file.

File: src/component/AccountIncomeStatus.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+                    <Image source={require('../../assets/images/camera.png')} />
```
Comment (msssoftprodigy008) [inline src/component/AccountIncomeStatus.js:18]: Please move all image imports to a single file.

File: src/component/TabView.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    const tabdata = ["Today", "Week", "Month", "Year"];
+    const [selectedTab, setSelectedTab] = useState(tabdata[0]);
+    const [filteredTransactions, setFilteredTransactions] = useState([]);
+
+    const currentDate = new Date();
+    const todayDate = currentDate.toISOString().split('T')[0];
+
+    const filterListByTab = (tab) => {
+        switch (tab) {
+            case 'Today':
```
Comment (msssoftprodigy008) [inline src/component/TabView.js:24]: Please move all constant values to a separate file.

constant.js
const TODAY = "Today"

component file
import {TODAY} from 'path_to_file'

File: src/component/UpperComponent.js [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+ * @param {number} props.totalIncome - The total income amount.
+ * @param {number} props.totalExpense - The total expense amount.
+ */
+
+const UpperComponent = ({ totalIncome, totalExpense }) => {
+    // Get the current date
+    const currentDate = new Date();
+
+    // Get the day of the week
+    const daysOfWeek = ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
```
Comment (msssoftprodigy008) [inline src/component/UpperComponent.js:21]: Move all constant value to constant file

File: src/component/UpperComponent.js [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    const currentDay = daysOfWeek[currentDate.getDay()];
+
+    // Get the date and month
+    const currentDateNumber = currentDate.getDate();
+    const monthsOfYear = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
```
Comment (msssoftprodigy008) [inline src/component/UpperComponent.js:26]: Move all constant value to constant file

File: src/redux/reducer.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
+const initialState = {
+    transactions: [],
+    selectedMonth: "month",
+    selectedCategory: "All"
+
+};
+
+const transactionsReducer = (state = initialState, action) => {
+    switch (action.type) {
+        case 'ADD_TRANSACTION':
```
Comment (msssoftprodigy008) [inline src/redux/reducer.js:10]: Create a separate action-type file for handling all cases.

action_type.js
const ADD_TRANSACTION = "ADD_TRANSACTION"

File: src/screens/AddTransactionScreen.js [hunk 9]
```diff
@@ -1,70 +1,116 @@
...
-        { label: 'Item 2', value: '2' },
-        { label: 'Item 3', value: '3' },
-        { label: 'Item 4', value: '4' },
-        { label: 'Item 5', value: '5' },
-        { label: 'Item 6', value: '6' },
-        { label: 'Item 7', value: '7' },
-        { label: 'Item 8', value: '8' },
+    const dispatch = useDispatch();
+
+    const categoryList = [
```
Comment (msssoftprodigy008) [inline src/screens/AddTransactionScreen.js:18]: Move all constant value to constant file

File: src/screens/AddTransactionScreen.js [hunk 10]
```diff
@@ -1,70 +1,116 @@
...
+        { label: 'Shopping', value: 'shopping' },
+        { label: 'Salary', value: 'salary' },
+        { label: 'Food', value: 'food' },
+        { label: 'Outing', value: 'outing' }
+    ];
+
+    const descriptionList = [
```
Comment (msssoftprodigy008) [inline src/screens/AddTransactionScreen.js:25]: Move all constant value to constant file

File: src/screens/Profile.js [hunk 11]
```diff
@@ -20,14 +20,17 @@ const Profile = () => {
...
           <Text style={styles.optionText}>Settings</Text>
         </TouchableOpacity>
+        <View style={styles.divider}></View>
         <TouchableOpacity style={styles.option}>
           <Image source={require('../../assets/images/exportData.png')} />
           <Text style={styles.optionText}>Export Data</Text>
         </TouchableOpacity>
+        <View style={styles.divider}></View>
         <TouchableOpacity style={styles.option}>
           <Image source={require('../../assets/images/logout.png')} />
```
Comment (msssoftprodigy008) [inline src/screens/Profile.js:35]: Move image to image file

**Review by msssoftprodigy008 — CHANGES_REQUESTED** [review]

File: src/component/TabView.js (hunk 5 above)
Comment (sarthak-sps) [reply src/component/TabView.js:25]: Implemented

**Pull request merged**